### ISSUE TITLE
Tapping on nodes that are part of a way does not present the Quest UI

### ISSUE THREAD

**Issue by tordans** [wtimme/OSM-Completionist#61]
Node: https://www.openstreetmap.org/node/433727998

This is about the bus shelter challenge.
- I have it active
- I tab the given node
- Nothing happens
- To double check, I tab a different challenge (handrail) - which works

![1 challenge](https://user-images.githubusercontent.com/111561/76159137-a6497800-611d-11ea-9a1c-273797a219c4.jpg) (the one that is inactive here, but active for the test)
![2 tab no action](https://user-images.githubusercontent.com/111561/76159136-a5b0e180-611d-11ea-805d-0b04c6eb6b17.jpg)
![3 tab on other challenges does show action](https://user-images.githubusercontent.com/111561/76159135-a47fb480-611d-11ea-96fd-a9737465aeac.jpg)

**Comment by wtimme** [comment]
Thanks for the report and the screenshots, @tordans! It was very helpful that you provided the link to the node. That way, I was able to take a look myself.

Indeed, at first it seems as if that particular bus stop was not selectable. However, if you long-tap, you'll see that there are quite some objects that can be selected:

![moeckernbruecke-long-tap](https://user-images.githubusercontent.com/1681085/76160021-3e476180-611e-11ea-87c2-3f3d5a76bccf.png)

I had a quick look at iD (to see if there was an issue with the data), but it looks fine to me. I think the culprit here is the way that the app figures out what it think you tapped. If you select the last option in the list that is presented upon long-tap, the quest dialog shows up as expected:

![moeckernbruecke-quest-dialog](https://user-images.githubusercontent.com/1681085/76160042-7babef00-611e-11ea-8b9e-6156b86e5409.png)

Maybe it makes sense to prioritize quest-enabled objects for the tap. What do you think?

**Comment by wtimme** [comment]
After a bit more digging, I realized that the issue is the lack of support for Quest-related nodes that are part of a way. I was not able to figure out a quick solution for this without introducing a lot of messy code.

**Comment by bryceco** [comment]
Yes, if you want to directly select a node that is part of a way without first selecting the way then it’s going to be messy. You can probably update the hit-testing code to do what you want, but you you’ll want to ensure you select the (correct) way it belongs to simultaneously. If you don’t then you could easily break assumptions made in other parts of the code.
